server/services/canvas/__init__.py

```diff
@@ -32,25 +32,43 @@ def get_user(canvas_id, key=None) -> FakeUser | User:
 def get_course(canvas_id, key=None) -> FakeCourse | Course:
     return _get_client(key).get_course(canvas_id)
 
+def is_staff_enrollment(enrollment_type: str):
+    return enrollment_type.lower() in ('ta', 'teacher')
 
 def is_course_valid(c) -> bool:
+    # A valid course has a name and id
+    # TBD if we should filter on published
     return not (not c) and \
         hasattr(c, 'id') and \
         hasattr(c, 'name') and \
-        hasattr(c, 'course_code') and \
-        hasattr(c, 'start_at') and \
-        hasattr(c, 'start_at_date')
+        hasattr(c, 'course_code')
+
+def noramlize_course_start_date(course: FakeCourse | Course) -> None:
+    # Ensure a valid start_at date for a course.
+    # return the term start_at_date if present
+    if course['term'] and course['term']['start_at']:
+        start_at = course['term']['start_at']
+        start_at_date = course['term']['start_at_date']
+    else:
+        start_at_date = course['start_at_date'] or course['created_at_date']
+        start_at = course['start_at'] or course['created_at']
+    course['start_at_date'] = start_at_date
+    course['start_at'] = start_at
 
 
 def get_user_courses_categorized(user: FakeUser | User) \
         -> tuple[list[FakeCourse | Course], list[FakeCourse | Course], list[FakeCourse | Course]]:
-    courses_raw = user.get_courses(enrollment_status='active')
+    courses_raw = user.get_courses(enrollment_status='active', include=['term'], per_page=100)
```

Comment:
would need to change fake canvas client to support `include` keyword

Reply:
oh does it not? I thought it did.. :( 
https://canvasapi.readthedocs.io/en/stable/keyword-args.html?highlight=include#list-parameters

Reply:
The real one does.
Our fake one does not.
I will do it.